### ISSUE TITLE
🚀   Elementor v3.5.0 Beta 6 Release  🚀 

### ISSUE THREAD

**Issue by DorShahar** [elementor/elementor#16840]
# [Beta 6 UPDATED - PLEASE READ BELOW!](https://github.com/elementor/elementor/issues/16840#issuecomment-966182585)

![Cover for post](https://user-images.githubusercontent.com/19154938/140638133-fba09273-f6a4-405e-93a5-46314549bc81.png)

Hi Elementor Beta Testers,
 
We are incredibly excited to introduce our new beta version, Elementor v3.5, which is all about increasing your design capabilities and improving your workflow. 

The new design features will allow you to elevate the appearance of your website and achieve visual effects that previously would have required additional coding or 3rd party plugins. 

With the workflow improvements, you will be able to shave off significant time from your project’s turnaround by improving productivity and eliminating redundant tasks. 

Try out Elementor v3.5 Beta and let us know what you think about our newest additions, including: 
 
- CSS Transform
- Text Stroke
- Word Spacing Control 
- Mark Widgets as Favorites
- Multi-Select
- Drag from Desktop 
- New Experiments UI

**Before you test, please make sure you are using:**
- Elementor v3.5.0 Beta 6
- Elementor Pro v3.5.2

## Enhancing Styling Capabilities 

![Group 1088045](https://user-images.githubusercontent.com/19154938/140638284-b7511043-4ceb-42e2-8fc8-20a13f5b05ce.png)

### Meet CSS Transform - Animate any Element 
![CSS Transform](https://user-images.githubusercontent.com/19154938/140638199-ede00210-be77-4e76-a114-03f6f804c745.png)

Would you like to rotate any element on your page on any axis, or enlarge an element upon hover to achieve highly engaging and trendy effects, but you don’t know CSS? 
No problem!
From now on, you can rotate, scale, skew, offset and flip any element on your page, within Elementor, without requiring a single line of code, that way - you gain complete control over your designs and achieve capabilities that weren’t available before. 
 
To take your element design to the next level, you can pair CSS Transform with our Pro feature - Motion Effects, to create incredible designs, without a single line of CSS or an extra plugin.

**Following is a list of CSS transformations available:**
- Rotate - 2D & 3D rotation: pivot element in any dimension
- Scale - Increase or decrease the element’s size 
- Skew - Transformation that skews an element
- Offset - Translate an element vertically or horizontally
- Flip - Create a mirror effect of the element - vertically or horizontally

**Here’s how it works:**

1. Create a new page, edit it with Elementor, and drag & drop any widget
2. Go to Advanced > Transform
3. Play around with the different options and make sure everything works as expected
4. If you are a Pro user:
a. Make sure you updated to Elementor Pro 3.5.0
b. Apply scrolling effects to this widget via Advanced > Motion Effects  > Scrolling effects 
c. Play around with the different options and make sure everything works as expected with motion effect as well

### Text Stroke
Text Stroke is a styling property that allows you to add a text stroke (outline) to any heading you wish to. 

Adding Text Stroke allows you to highlight specific elements on the page by giving them an outline. With this styling option, you can set the font to transparent, and use text stroke to create an outline, or you can choose a color for the text, and a separate color for the outline to make your design stand out. 

Text stroke will be available for the following widgets: Icon Box, Tabs, Accordion, and Text Path. It will also be available for the following Pro widgets in the upcoming Elementor Pro v3.6: Posts, Animated Headline, Price List, Price Table, Flip Box, Call to Action, Testimonial Carousel, and Countdown.

**Here’s how it works:**
1. Create a new Page, edit it with Elementor and drag & drop Heading widget
2. Go to the widget’s panel, and under Style > Title click on Text Stroke control
3. Play around with the different options and make sure everything works as expected
4. This feature is also supported in the Icon Box widget, Tabs widget, Accordion widget, and Text path widget
5. Make sure everything works as expected both on Frontend and in the Editor

### Word Spacing Control
Similar to Letter Spacing, the new Word Spacing control in the typography group control allows you to set the distance between one word and another. Doing so enables you to increase your design customization of textual elements, including but not limited to headings, text editors, buttons, and more.

** Here’s how it works:**

1. Under any widget that contains text, go to “Style” tab
2. Click the Pencil “edit” icon of the Typography control
3. Change the space between words using the “Word spacing” control
4. Use this setting in a Global Font
5. Make sure everything works as expected both on Frontend and in the Editor

## Improving Workflow and Productivity 
![Main Workflow](https://user-images.githubusercontent.com/19154938/140639822-3b25912b-7c96-492b-b045-73c5b6d25d6e.png)

### Meet Favorite Widgets - Save Precious Time and Pin your Favorite Widgets
![Mark widgets as favorites](https://user-images.githubusercontent.com/19154938/140638462-1f02adb4-5d91-488e-9324-38d73e11644e.png)

As you tend to generate an affinity towards specific widgets and use them often, we’ve created a dedicated ‘Favorites’ section in the Editor Panel for you to pin your favorite widgets to. While we always encourage you to incorporate new widgets in your websites, when they are appropriate, pinning widgets to your ‘Favorites’ section will allow you to work faster by personalizing the Editor to you. 

Once you mark widgets as your favorite, they will appear in a dedicated ‘Favorites’ section. Widgets added to this section will appear in alphabetical order.

**Here’s how it works:**

1. Edit a page with Elementor
2. Open the widgets panel
3. Right click on any widget
4. Click “Add to Favorites”
5. The widget will now appear at the top of the panel under a dedicated “Favorites” section 
6. Go to a different page or post in your site and make sure the settings has been saved for your user
7. Switch user and see that the settings are relevant only for your user
 
### Multi-Select - Work Faster and Impact Several Elements at Once 
![Multi Select](https://user-images.githubusercontent.com/19154938/140638498-4d73e6c1-d85f-4e37-8315-d29411c06bea.png)

As you build out your websites, you may notice that the style of some sections, columns or widgets repeat themselves throughout the page, or that you would like to copy, or delete a number of them simultaneously. To improve your workflow and allow you to scale your website creation, you can leverage multi-select to impact multiple elements at once; including: copy, delete, duplicate, paste, paste style, and reset style. 

For example, by clicking ctrl (or CMD in Mac) while clicking a number of widgets, columns or sections in the editor or the navigator, you can delete them instantly, instead of deleting them one by one. Another example is when you have a number of similar elements, such as button widgets in a column, and you want to make them all in one row, you can style one of them to be in-line, and instantly copy that style to all the other buttons, at once. 

**Here’s how it works:**

1. Edit a page using Elementor
2. Select any page element
3. Press the ctrl key (CMD key on mac)
4. Select additional page elements in the live preview and / or the navigator panel
5. Supported actions will apply for the entire selection

### Drag from Desktop - Improve your Workflow by Dragging Media Files Directly into the Editor
![Drag from Desktop](https://user-images.githubusercontent.com/19154938/140638594-5b7820aa-cb27-4ae6-9dec-a9008f0d5c90.png)

Skip the: “upload to media library,” drag in a widget ,and select the media file steps, and simply drag media files from your desktop directly onto the page. You can drag one or several media files to the editor, and have them all appear in one section, making your workflow easier, and more efficient. 

Dragging a media file onto the page will automatically add the media files to the media library, and create the correlating widget on the page (a dragged image will automatically place an image widget in the section, and a video will automatically place a video widget in the section).

In addition to media files, you’ll be able to easily drag any saved templates files (.json) onto your website.
 
**Here’s how it works:**

1. Edit a page with Elementor
2. Drag one or multiple images from your desktop into the screen
3. Repeat the same step for videos and exported Elementor template files (.json)
4. The assets will be uploaded to the media library and added to the page as the corresponding widgets

### Tweak: Improved Experiments UI 
The improved Experiments user interface makes it easier for you to understand the maturity and status of each experiment. The top section of experiments displays those that are ongoing, and displays their status - alpha or beta, whereas the bottom section displays those that are more stable. 

With the new interface, you can also activate and deactivate all experiments in a single-click. 

**Here’s how it works:**

1. Go to the WP Dashboard > Elementor > Settings > Experiments tab
2. Activate some experiments
3. Click on Deactivate all experiments button. Make sure all the experiments are in Inactive status
4. Go to the Editor and make sure there isn’t an experiment in an active state (for example no Additional Custom Breakpoints, no Color Sampler, etc’)
5. Go back to WP dashboard and make sure there isn’t an experiment in an active state (for example no Import Export template)
6. Go back to the Experiments tab and click on Activate all experiments. Make sure all the experiments are in the Active status
7. Repeat steps (4) and (5) and this time make sure the experiments are working properly

### Eicons library is now supported in the inline SVG experiment 
This new experiment achieves faster page load by changing the loading mechanism of eicons into inline SVG. Keeping up with our commitment to simplifying the process and making constant performance improvement so your website can load rapidly, with this experiment turned on, we reduce CSS and Woff size on each page load by up to 111KB. Read more about [how Elementor improved asset loading](https://developers.elementor.com/how-elementor-improved-asset-loading-and-made-your-website-run-faster/).

**Here’s how it works:**
1. Make sure the Inline Font Icons experiment is Inactive in Settings > Experiments
2. Create a new page, Drag & Drop an Image widget, and choose any image.
3. Under the Content tab, set Link to Media file and Lightbox to Yes.
4. Preview the page, click on the image to launch the Lightbox. Make sure it works.
5. In developers tool, under the network tab, look for “eicons” files. Make sure they appear. 
6. Go back to the Experiments tab, and activate Inline Font Icons experiment
7. Reload the page you recently created. 
8. In the developers tool, make sure all assets related to font Awesome and eicons were not loaded
9. Make sure everything is working as expected with various widgets that contain icons both on Frontend and Editor.

## More Improvements
- New: Meet Multi-Select for page elements - improve your workflow process by making bulk actions (#8006, #879)
- New: Added Text Stroke control - highlight titles with colorful outlines to in Heading widget, Icon Box widget, Tabs widget, Accordion widget, and Text path widget

- New: Meet Favorite widgets section - save your most useful widgets for easier accessibility (#2184,#11443)
- New: Introducing CSS Transform - rotate, scale, skew, offset and flip any element  (#12451)
- Experiment: Hide native WordPress widgets from search results - remove unwanted widgets from the widgets panel
- Tweak: Meet the new experiments UI - find easily experiments 
- Tweak: Added Inline Font Icons experiment to work with `eicons` library (#8572, #11361)
- Tweak: Added Elementor price plan filter to Kit Library (#16075)
- Tweak: Added reverse columns to Additional Custom Breakpoints Experiment (#16322, #12925)
- Tweak: Added the option to search by tag names in Kit Library (#16075)
- Tweak: Deprecated `get_add_new_landing_page_url` (props [@HeyMehedi ](https://github.com/HeyMehedi))
- Tweak: Added word spacing control to typography settings
- Tweak: Added promotion for Custom Code feature
- Tweak: Removed the Archive Posts and Archive Title widgets from widgets panel search in non-archive templates
- Tweak: Extend "Edit Buttons" options for 3rd party developers 
- Tweak: Optimize Kit library index page performance
- Tweak: 'Editor/Documents' moved to `/editor/components` folder
- Tweak: Prompt the user permission to allow unfiltered file uploads in the Import Kit flow
- Tweak: Added `filter` to Context Menu Groups
- Tweak: Preparation for Pro feature
- Tweak: UI improvement for cascaded control units
- Tweak: Added shortkeys description to the tooltips in the footer of the editor panel
- Tweak: Added a quick Apply Kit modal option in the Kit Library
- Tweak: Improved performance of `WP_Query` by adding `no_found_rows=true` parameter
- Tweak: Improved performance of loading the "My templates" while loading the Templates library
- Tweak: Minimum height wasn't defined to all responsive devices
- Tweak: Experiments status added to the system info for better debugging
- Tweak: Added Safe mode support for Elementor Experiments
- Tweak: Updated Dialog library to 4.9.0
- Tweak: Added a unified registration for components for widgets, controls, form actions, dynamic tags, and finder categories
- Tweak: Added support for future feature compatibility
- Tweak: Disable the option to change the desktop height and width fields in the responsive bar
- Tweak: Detect and present errors in the Import / Export flow for better troubleshooting
- Fix: Hamburger button didn't open on mobile in Nav menu widget (#16682)
- Fix: Possible conflict between Jet-Engine and Elementor Form widget (#13117)
- Fix: Finder incorrectly identifies pages created through Finder as a Post type instead of Page type (#12502)
- Fix: Unexpected behavior if `ctrl` + mouse click was used on mac (#5228)
- Fix: Edit header handles don't show up if the `z-index` of the header section is above 99 (#16214)
- Fix: Dividers not vertically centered in Icon List widget (#16431)
- Fix: SVG sanitizer is failing if there is a line break after `</svg`
- Fix: Slides per view can't be set to default if there is a value in widescreen in the Testimonial Carousel widget (#16210)
- Fix: Image Carousel widget ignored padding and margin inside columns
- Fix: Theme Builder close button not functioning if the page was refreshed
- Fix: Pro widget promotional popover CTA text was incorrect when Elementor Pro is activated but not connected
- Fix: Section editing handle is not accessible when Section is placed below a Header
- Fix: Run DB calculations only on sites that are required to
- Fix: Updated Admin Top Bar font for Non-English languages
- Fix: UI glitch in Google maps widgets
- Deprecated: See all deprecations to this version in our [Developers Deprecations Post](https://developers.elementor.com/v3-5-planned-deprecations/)

## Make an Impact: Share Feedback Before Features Are Released 
We’re always looking to get to know Elementor customers on a deeper level. If you’re interested in meeting with our team and sharing your insights, feedback, and feature requests you’re welcome to sign up for some quality time with us [here](https://uxpractices.elementor.cloud/joinresearch). Or download [Elementor’s Dev Edition](https://wordpress.org/plugins/elementor-beta/). Talk soon! 

---
## How to Install the Beta Version

- If you still haven't enabled beta testing in Elementor, please follow these steps:
- Go to Elementor > Tools > Versions.
- Enable the beta tester feature.
- Go to the Dashboard > Updates and update to the latest beta version.
- You can also download the new version here: https://downloads.wordpress.org/plugin/elementor.3.5.0-beta8.zip

---
Notes
For bugs and remarks, please reply with a comment on this issue here on GitHub.
Please keep this related to the current beta only! For any other issue or feature request, feel free to open a new issue.
Designated time for release: `December 15, 2021 (Estimated)`

Many thanks for your support and help!
Cheers,
_Elementor Team_

**Comment by 79ho3ein** [comment]
excellent. It would be great if the transform feature worked for sections and columns as well. Currently only works on widgets.
Overall everything is great. Great job

**Comment by rodolphebertozzo** [comment]
Please see : https://github.com/elementor/elementor/issues/16793

**Comment by DorShahar** [comment]
Hi @rodolphebertozzo  - thank you for reporting, we will investigate this on

---

@79ho3ein  - thank you for your feedback. There is no plan to apply transformation for columns and sections since they are the main blocks for the page layout, and other elements are relative to them. 

As for the inner section: we will investigate.

**Comment by fabianenzensperger** [comment]
Hi! Yes, that's all very nice.

CSS transform and text stroke are very useful for the design, but the other things are probably not that important in my eyes and especially not at this point in time. But still nice that Elementor is being improved in small steps in some places.

There is a video from Elementor that I think was released in the summer with an announcement of features that should appear in 2021. Will we actually see all of the announced features in Elementor in 2021? I think time is getting scarcer, especially considering the beta lead time and only one milestone is removed from the list in this version.

<img width="372" alt="Bildschirmfoto 2021-11-07 um 23 05 54" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">


What can already be seen in Dev or Beta Edition is marked in bold and what is already included in the official release is ticked off. There is still a lot (important) missing.


![FCJlfUZXsAkXG8l](https://user-images.githubusercontent.com/57997262/140663631-d046dbe6-ca7e-4cac-9570-952ec45c2ef1.jpeg)

You also posted this picture on Twitter. Is that a serious picture and if so, when is this new surface planned for?
I have to say, I like the (supposedly) new interface very much, it leaves some problems that have been solved a bit cumbersome so far, such as the left editor column, which can be very annoying when you are working on the design, and the responsive bar does not always have to be faded in and out with great effort.

Thank you in advance for your feedback and wish you a good start to the week.

**Comment by DorShahar** [comment]
Hi @fabianenzensperger 
Thank you for your comment. 
You marked that performance as an unchecked checkbox. 
Well, we are constantly improving our performance. You can see in the following blog posts and Beta posts how we improved our infrastructure:

1. [Additional Custom Breakpoint](https://developers.elementor.com/additional-custom-breakpoints-technical-details-and-gotchas/#:~:text=Big%20Performance%20Improvements) - the reason it took a while to develop it since we didn't compromise in performance, and we improved up to 23% in page load time and we reduce the widgets and controls data load by about a third (from 900KB to 600KB). 
2. [Improved Assets loading experiment](https://developers.elementor.com/how-elementor-improved-asset-loading-and-made-your-website-run-faster/) - load only the assets that are in use on the page. 
3. [Improved CSS Loading experiment ](https://github.com/elementor/elementor/issues/15336)- in v3.3.0 we added a new experiment that saved 120KB in Core and 225KB in Pro
4. [ Optimize DOM experiment](https://developers.elementor.com/elementor-v3-2-additional-dom-improvements/) - we are constantly improving our DOM structure

We encourage you to activate the experience and improve your page performance. 
In addition, we do not stop improving our page performance in general, and focusing on performance in every single feature that we are releasing. 

Thanks

**Comment by Michaelgimii** [comment]
hi @DorShahar, Will neew editor ui be released in 2021?

**Comment by DorShahar** [comment]
Hi all
The new UI is under development and it's only a POC at the moment. 
Please keep this thread only for issues related to the beta.

Thanks

**Comment by rodolphebertozzo** [comment]
Hi @DorShahar, 
Wasn't the new element container planned for this version normally?

**Comment by fabianenzensperger** [comment]
> Hi @fabianenzensperger Thank you for your comment. You marked that performance as an unchecked checkbox. Well, we are constantly improving our performance. You can see in the following blog posts and Beta posts how we improved our infrastructure:
> 
> 1. [Additional Custom Breakpoint](https://developers.elementor.com/additional-custom-breakpoints-technical-details-and-gotchas/#:~:text=Big%20Performance%20Improvements) - the reason it took a while to develop it since we didn't compromise in performance, and we improved up to 23% in page load time and we reduce the widgets and controls data load by about a third (from 900KB to 600KB).
> 2. [Improved Assets loading experiment](https://developers.elementor.com/how-elementor-improved-asset-loading-and-made-your-website-run-faster/) - load only the assets that are in use on the page.
> 3. [Improved CSS Loading experiment ](https://github.com/elementor/elementor/issues/15336)- in v3.3.0 we added a new experiment that saved 120KB in Core and 225KB in Pro
> 4. [ Optimize DOM experiment](https://developers.elementor.com/elementor-v3-2-additional-dom-improvements/) - we are constantly improving our DOM structure
> 
> We encourage you to activate the experience and improve your page performance. In addition, we do not stop improving our page performance in general, and focusing on performance in every single feature that we are releasing.
> 
> Thanks


Hey @DorShahar,

you are definitely right and the performance is actually noticeable! I am aware of these things, I just didn't tick it off because there were always improvements over the year and one or the other thing has not yet been completed or there could be further improvements right now with regard to "Div / Container Widget", "Flexbox Container" and "Nested Elements".

That should again ensure performance, I think. 🤗

**Comment by rodolphebertozzo** [comment]
> Hi! Yes, that's all very nice.
> 
> CSS transform and text stroke are very useful for the design, but the other things are probably not that important in my eyes and especially not at this point in time. But still nice that Elementor is being improved in small steps in some places.
> 
> There is a video from Elementor that I think was released in the summer with an announcement of features that should appear in 2021. Will we actually see all of the announced features in Elementor in 2021? I think time is getting scarcer, especially considering the beta lead time and only one milestone is removed from the list in this version.
> 
> <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">
> 
> What can already be seen in Dev or Beta Edition is marked in bold and what is already included in the official release is ticked off. There is still a lot (important) missing.
> 
> ![FCJlfUZXsAkXG8l](https://user-images.githubusercontent.com/57997262/140663631-d046dbe6-ca7e-4cac-9570-952ec45c2ef1.jpeg)
> 
> You also posted this picture on Twitter. Is that a serious picture and if so, when is this new surface planned for? I have to say, I like the (supposedly) new interface very much, it leaves some problems that have been solved a bit cumbersome so far, such as the left editor column, which can be very annoying when you are working on the design, and the responsive bar does not always have to be faded in and out with great effort.
> 
> Thank you in advance for your feedback and wish you a good start to the week.

Hi @fabianenzensperger, I agree with your state of mind!
Indeed, after thinking about it for a long time, the classic editor is not bad in itself, but not clean enough, and to come to a new editor with what seems to be a floating bar to manage the options of each element and a floating window that opens when you want to add the elements (and I refer to what I see on the picture) seems to me to be something extraordinary and magical. @DorShahar, @shilo-ey  In any case, you have all my support for the wonderful work you are doing, and I will continue more than ever, maybe less in beta but directly in the development versions to test as soon as possible each of your features and to give you my feedbacks.

**Comment by black-eye** [comment]
Hi  @DorShahar, based on the post here: https://developers.elementor.com/v3-5-planned-deprecations/ it seems that Elementor 3.5 has introduced some fundamental changes in managing the widgets and controls (registering, unregistering, etc.). Many methods related to creating the new widgets are now softly deprecated since 3.5. 

The problem is, that the developers docs still doesn't reflect this change (see: https://developers.elementor.com/add-custom-functionality/#Registering_New_Widgets). The new method `register()` is also nowhere to be found there (nor the explanation on how to use it).

Could you please clarify us these new changes and how do they affect creating custom widgets and controls? Thanks.

**Comment by 79ho3ein** [comment]
In connection with Multi-Select ...
If I select elements to delete directly **without activating the navigation panel**, I feel there is interference.
Once it works right, once it does not work right! But I do not experience this problem in paste style and it always works properly. There is only difficulty in selecting to delete elements.  Friends, please try this

https://user-images.githubusercontent.com/77163248/140827136-dde9a0ef-5260-4ba8-8e26-7066a041a7c0.mp4

**Comment by fabianenzensperger** [comment]
Hi @DorShahar and @shilo-ey,

### Thoughts on text stroke and border

In the (for example) design and prototype tool "Sketch" and also in the Affintity / Adobe programs there is the following function:

<img width="240" alt="Bildschirmfoto 2021-11-09 um 00 26 40" src="https://user-images.githubusercontent.com/57997262/140834358-ec96b9df-e853-4140-95a5-bc5d4a0609a9.png">

Namely, the possibility of displaying the frame or the text stroke (which is nothing more than a frame) "inside", "center" and "outside". I think something like that should be adjustable with this function, because most of the time the text stroke or the frame in Elementor looks very curious and this is probably due to the standard alignment "center" that Elementor uses.

I would like some feedback on that. Because this could also be integrated into Elementor for all types of border or stroke.

### Suggestion to favorites

<img width="299" alt="Bildschirmfoto 2021-11-09 um 00 36 34" src="https://user-images.githubusercontent.com/57997262/140835174-8e9490a5-435f-4639-8407-9e7a968b1bd9.png">


And in addition to the favorites function, I also notice that this group, like the search function, should be pinned to the top. So adhesive, so that you don't have to scroll back up to find your favorite widget. Good idea?

**Comment by fabianenzensperger** [comment]
### CSS Transform problem  in Editor

Transform a button in the editor is a little bit buggy. In Front end it looks "good" (of chourse not, because it's a test with weird combination of tranforms 😅).

Editor:

<img width="1077" alt="Bildschirmfoto 2021-11-09 um 00 44 25" src="https://user-images.githubusercontent.com/57997262/140836172-ccf12c61-fbba-4754-be0f-31d0e68c251b.png">


Frontend:

<img width="443" alt="Bildschirmfoto 2021-11-09 um 00 46 43" src="https://user-images.githubusercontent.com/57997262/140836176-8abf54db-62e4-4ecf-bb77-c79b98ee32f4.png">

Maybe you should test the individual behavior of the various widgets again, because I haven't checked all of them. 👀

**Comment by Kishorchandth** [comment]
Bug:

1. After enabling Font-Awesome Inline eicons (Hamburger menu icon) were inline properly but Close Icon was not rendering.
![Close Icon is not rendering](https://user-images.githubusercontent.com/39625343/140870792-7b15ba78-48af-4510-ae88-0575b8455350.jpg)

2. Tweak: Improved Experiments UI is great but when experiments get bigger we have to scroll more. I would request **save changes** to be sticky to the top or bottom.

**Comment by ramiy** [comment]
> Hi @DorShahar, based on the post here: https://developers.elementor.com/v3-5-planned-deprecations/ it seems that Elementor 3.5 has introduced some fundamental changes in managing the widgets and controls (registering, unregistering, etc.). Many methods related to creating the new widgets are now softly deprecated since 3.5.
> 
> The problem is, that the developers docs still doesn't reflect this change (see: https://developers.elementor.com/add-custom-functionality/#Registering_New_Widgets). The new method `register()` is also nowhere to be found there (nor the explanation on how to use it).
> 
> Could you please clarify us these new changes and how do they affect creating custom widgets and controls? Thanks.

We are planning to release the new developers docs before the final release of Elementor 3.5. You will find there all the changes, with new tutorials. Meanwhile, check the [deprecations post](https://developers.elementor.com/v3-5-planned-deprecations/) in the developers blog.

Also, keep in mind that deprecated methods won't break your code. Soft deprecation last 4 major version, then you have another 4 major version of hard deprecation. Only after 8 version we delete the old code. Theoretically, you can use the old methods until Elementor 4.3.0.

**Comment by DorShahar** [comment]
Hi @momo-fr, thanks for reaching out. Can you share a video that explains why it is difficult to use? 

--- 

Hi @79ho3ein , thank you for your feedback. However, I couldn't reproduce the scenario in your video. 
If someone else experiencing this, please reach out. 

--- 

@fabianenzensperger , as for your comments:
1. Text stroke & border : we will think about this. However, we cannot implement suggestion like this during the Beta phase.
2. Pined the search bar: nice suggestion! we will discuss it. 
3. Transform (skew) - I wasn't able to reproduce it on my end. If anyone else  experiencing this, please reach out with a video. 

---

Hi @Kishorchandth , thank you for checking our beta.
1. We are aware of the bug with the  close button in the Nav menu widget , and we will fix it in the next beta.
2. At the moment we will not change the location of the button. By using the buttons of Activate and Deactivate all experiments you don't need to click on the "save" button

Thanks again to everyone.

**Comment by momo-fr** [comment]
@DorShahar, [see here](https://screencast-o-matic.com/watch/crXQQlVlLyH), if we can select the origin point of the element (in my case top/left) we can rotate title correctly in editor i think.

Another point seems to me to be missing for drag & drop function in the editor: being able to drag a plain text file (.txt).

**Comment by DorShahar** [comment]
@momo-fr  - thank you for the video. There is no bug. 
You can control the anchor point using the relevant controls, as you can see below
<img width="301" alt="transform_anchor_point" src="https://user-images.githubusercontent.com/19154938/140907776-49a335b1-0140-45ec-bfa5-55248b429b57.png">

As for D&D a `.txt` file - what is the use case? How do you see this implemented (for example the text should be added to the page as a Text editor widget? Heading widget? A button to download this `txt` file? ) 

Thanks.

**Comment by momo-fr** [comment]
Thank you @DorShahar, I have missing this settings.
For drag&drop txt file in Editor widget I think.

**Comment by fabianenzensperger** [comment]
> Hi @momo-fr, thanks for reaching out. Can you share a video that explains why it is difficult to use?
> 
> Hi @79ho3ein , thank you for your feedback. However, I couldn't reproduce the scenario in your video. If someone else experiencing this, please reach out.
> 
> @fabianenzensperger , as for your comments:
> 
> 1. Text stroke & border : we will think about this. However, we cannot implement suggestion like this during the Beta phase.
> 2. Pined the search bar: nice suggestion! we will discuss it.
> 3. Transform (skew) - I wasn't able to reproduce it on my end. If anyone else  experiencing this, please reach out with a video.
> 
> Hi @Kishorchandth , thank you for checking our beta.
> 
> 1. We are aware of the bug with the  close button in the Nav menu widget , and we will fix it in the next beta.
> 2. At the moment we will not change the location of the button. By using the buttons of Activate and Deactivate all experiments you don't need to click on the "save" button
> 
> Thanks again to everyone.


@DorShahar Thanks for the answers. Should I open a thread with this feature request or have you already made a note of it?

**Comment by DorShahar** [comment]
@fabianenzensperger  - please open and connect to this issue, so it will be easier for us to track. 
Thanks again

**Comment by 79ho3ein** [comment]
I think it would be great if the Drag from Desktop feature also supports the lottie widget. Dragging the json file into the editor, adds the Lottie widget.

**Comment by fabianenzensperger** [comment]
> Hi @DorShahar,
> Wasn't the new element container planned for this version normally?

I know this is offtopic now. Although, is that it at all? The 3.5 update is not that exciting, cool, but not too exciting, so you can add a little spice here.. 🧂

But the latest dev-beta (3.6.0-dev2) finally has the **flexbox container**!! And I'll tell you, it's an absolute dream and so much fun. I think this is the next level. Finally with a sensible (responsive) column layout and the possibility for much more complex layouts. Simply world class. 😍😍😍

Really thank you, thank you, thank you, I'm really excited as soon as this is in the official beta. Maybe already in 3.6.0.. 👀

**Comment by sf-steve** [comment]
> > Hi @DorShahar,
> > Wasn't the new element container planned for this version normally?
> 
> I know this is offtopic now. Although, is that it at all? The 3.5 update is not that exciting, cool, but not too exciting, so you can add a little spice here.. 🧂
> 
> But the latest dev-beta (3.6.0-dev2) finally has the **flexbox container**!! And I'll tell you, it's an absolute dream and so much fun. I think this is the next level. Finally with a sensible (responsive) column layout and the possibility for much more complex layouts. Simply world class. 😍😍😍
> 
> Really thank you, thank you, thank you, I'm really excited as soon as this is in the official beta. Maybe already in 3.6.0.. 👀

@fabianenzensperger Is there a changelog somewhere for the dev edition? We write a lot of custom widgets, and we try to keep an eye on depreciations etc to make sure our code doesnt break, but i have never worked out how to keep track of upcoming features in dev.

**Comment by fabianenzensperger** [comment]
> > > Hi @DorShahar,
> > > Wasn't the new element container planned for this version normally?
> > 
> > 
> > I know this is offtopic now. Although, is that it at all? The 3.5 update is not that exciting, cool, but not too exciting, so you can add a little spice here.. 🧂
> > But the latest dev-beta (3.6.0-dev2) finally has the **flexbox container**!! And I'll tell you, it's an absolute dream and so much fun. I think this is the next level. Finally with a sensible (responsive) column layout and the possibility for much more complex layouts. Simply world class. 😍😍😍
> > Really thank you, thank you, thank you, I'm really excited as soon as this is in the official beta. Maybe already in 3.6.0.. 👀
> 
> @fabianenzensperger Is there a changelog somewhere for the dev edition? We write a lot of custom widgets, and we try to keep an eye on depreciations etc to make sure our code doesnt break, but i have never worked out how to keep track of upcoming features in dev.


I think you have to ask the two guys: @shilo-ey @DorShahar

**Comment by kangarko** [comment]
@shilo-ey I am unable to edit my header, cannot click any elements, this appeared in the console:

![a](https://i.imgur.com/QhDEpe1.png)

**Comment by webdevs-pro** [comment]
@shilo-ey Old bug with copy/paste styles when widget has custom repeater control in advanced tab still not fixed. For example Crocoblocks JetEnjine Dynamic Visibility feature adds repeater in advanced tab.


You can see errors on screenshot after paste style
![image](https://user-images.githubusercontent.com/14045233/141239112-7a5c82de-07f5-4f82-9fcd-fde50dda7911.png)

**Comment by shilo-ey** [comment]
@webdevs-pro Thank you, 
This is not the bug that was fixed. we implemented a fix for the multiple repeaters issue in the same tab. Have you looked for if this existing issue has already been reported? 

Please keep this thread to the beta scope solely.

Thanks

**Comment by DorShahar** [comment]
Hey all,
We just released a new beta version, please update your version to **Elementor v3.5.0 Beta 2.**
The new version includes the following: 
* Fix: Missing text typography control in Text Path widget (#16805)
* Fix: Activate and Deactivate all experiments buttons didn't work as expected in the Experiments screen (#16531)
* Fix: Inner Section widget is not working as expected when insert into a page (#16716)
* Fix: Custom width controls disappears in responsive devices when Additional Custom Breakpoint experiment is active in Position section
* Fix: CLI 'library import' deletes the original imported file when importing a template
* Fix: Remove redundant data attributes in CSS Transform
* Fix: JSON file imported to a wrong location when using the Drag from Desktop feature
* Fix: JSON file can't import more than once when using the Drag from Desktop feature
* Fix: Wrong default value in Load Font Awesome 4 support in Elementor advanced screen
* Fix: Wrong default values in CSS Print method in Elementor advanced screen
* Fix: SVG file was imported when choosing the skip option in the Enable Unfiltered Files Uploading modal when importing a kit
* Fix: Right-click menu didn't open when clicking the 'Add New Section' area in edit mode
* Fix: License page top bar is didn't work as expected in Elementor top bar
* Fix: Menu Anchor widget didn't scroll to the appropriate ID if Scroll Snap is active
* Deprecated: Removed Slick.js library. See all deprecations to this version in our Developers Deprecations Post
Fix: Template name changed if a dash mark was used in the import process

Please keep this related to the Current beta Only! any other[ Issue Report](https://github.com/pojome/elementor/issues/new?template=Bug_report.md) or [Feature Request](https://github.com/pojome/elementor/issues/new?template=Feature_request.md) feel free to open a new issue.

Thanks again to everyone!

**Comment by rodolphebertozzo** [comment]
Hi, can you review please ? ;) : https://github.com/elementor/elementor/pull/16905

**Comment by jazir5** [comment]
@DorShahar 

https://github.com/elementor/elementor/issues/6778

When are you guys going to finally add background image size selection? This is one of the most highly requested features on github, and we cannot get a response from you guys. How hard is it to add a dropdown menu for this? It already exists for the image widget, can you not just insert the code that already exists for image selection for background images? Can you please address this already? This is a massive performance issue.

I'm posting this here because otherwise you guys simply won't give this issue any attention.

**Comment by acal** [comment]
> @DorShahar
> 
> #6778
> 
> When are you guys going to finally add background image size selection? This is one of the most highly requested features on github, and we cannot get a response from you guys. How hard is it to add a dropdown menu for this? It already exists for the image widget, can you not just insert the code that already exists for image size selection for background images? Can you please address this already? This is a massive performance issue.
> 
> I'm posting this here because otherwise you guys simply won't give this issue any attention. It has been requested for years. You have clearly had a more performance focused mindset lately, yet you are completely ignoring one of the easiest ways to fix performance issues in Elementor. This issue cannot be hard to fix as the code for image selection already exists. This is probably one of the simplest performance issues to fix out of any that need to be addressed.

+1,000,0000,000

**Comment by jazir5** [comment]
@avivu @shilo-ey

**Comment by fabianenzensperger** [comment]
> @DorShahar
> 
> #6778
> 
> When are you guys going to finally add background image size selection? This is one of the most highly requested features on github, and we cannot get a response from you guys. How hard is it to add a dropdown menu for this? It already exists for the image widget, can you not just insert the code that already exists for image size selection for background images? Can you please address this already? This is a massive performance issue.
> 
> I'm posting this here because otherwise you guys simply won't give this issue any attention. It has been requested for years. You have clearly had a more performance focused mindset lately, yet you are completely ignoring one of the easiest ways to fix performance issues in Elementor. This issue cannot be hard to fix as the code for image selection already exists. This is probably one of the simplest performance issues to fix out of any that need to be addressed.


That sounds interesting, but usually an extra plug-in is required for this, which changes the file size in KB as well as in the dimensions when uploading. I don't know how Elementor could implement that, let alone whether it falls within Elementor's area of responsibility to control it. If I have not understood it, please explain it to me.

**Comment by acal** [comment]
> > @DorShahar
> > #6778
> > When are you guys going to finally add background image size selection? This is one of the most highly requested features on github, and we cannot get a response from you guys. How hard is it to add a dropdown menu for this? It already exists for the image widget, can you not just insert the code that already exists for image size selection for background images? Can you please address this already? This is a massive performance issue.
> > I'm posting this here because otherwise you guys simply won't give this issue any attention. It has been requested for years. You have clearly had a more performance focused mindset lately, yet you are completely ignoring one of the easiest ways to fix performance issues in Elementor. This issue cannot be hard to fix as the code for image selection already exists. This is probably one of the simplest performance issues to fix out of any that need to be addressed.
> 
> That sounds interesting, but usually an extra plug-in is required for this, which changes the file size in KB as well as in the dimensions when uploading. I don't know how Elementor could implement that, let alone whether it falls within Elementor's area of responsibility to control it. If I have not understood it, please explain it to me.

@fabianenzensperger perhaps I'm not informed enough but... since the image widget allows you to choose from one of the WordPress-generated thumbnail sizes per responsive break, then why can't Elementor's background image css controls use the same? I don't know of a plugin that extends this part of Elementor (please refer if you do)... but why is a plugin capable of this and not Elementor itself?

**Comment by sf-steve** [comment]
> That sounds interesting, but usually an extra plug-in is required for this, which changes the file size in KB as well as in the dimensions when uploading. I don't know how Elementor could implement that, let alone whether it falls within Elementor's area of responsibility to control it. If I have not understood it, please explain it to me.

For foreground images, you can select the image dimensions, either a predefined (core or theme/plugin) thumbnail size, eg medium, or custom. This will then use the cropped image. 
This allows a user to upload 1 large (1920*1080 200kb) image, to use as a background for example, and then select a small (400*200 40kb) crop somewhere else as a thumbnail. 

Background image sin elementor lack this feature, so the large image is always used, even if its only covering a small area

**Comment by jazir5** [comment]
> > @DorShahar
> > #6778
> > When are you guys going to finally add background image size selection? This is one of the most highly requested features on github, and we cannot get a response from you guys. How hard is it to add a dropdown menu for this? It already exists for the image widget, can you not just insert the code that already exists for image size selection for background images? Can you please address this already? This is a massive performance issue.
> > I'm posting this here because otherwise you guys simply won't give this issue any attention. It has been requested for years. You have clearly had a more performance focused mindset lately, yet you are completely ignoring one of the easiest ways to fix performance issues in Elementor. This issue cannot be hard to fix as the code for image selection already exists. This is probably one of the simplest performance issues to fix out of any that need to be addressed.
> 
> That sounds interesting, but usually an extra plug-in is required for this, which changes the file size in KB as well as in the dimensions when uploading. I don't know how Elementor could implement that, let alone whether it falls within Elementor's area of responsibility to control it. If I have not understood it, please explain it to me.

Its already implemented for the image widget. They basically just need to copy paste the code for background images. Its the exact same feature that is already implemented. Wordpress automatically creates multiple scaled image sizes(such as medium, medium large, large, etc) that are selectable via a dropdown for the image widget. This does not require any additional plugins as the feature already exists for the image widget.

**Comment by kangarko** [comment]
**@DorShahar @shilo-ey I am still unable to edit my header on the latest beta, did you miss my comment?**

![a](https://i.imgur.com/XCfyjto.png)

**Comment by DorShahar** [comment]
Hi everyone
1. As for [#6778](https://github.com/elementor/elementor/issues/6778) - this task is important to us. We are allocating our resources to handle and improve more areas in the product:
a. See my performance comment in this thread
b. The Container that we release to `dev-edition` that all of you mentioned
c. The Woo widgets improvement in Elemnentor Pro 3.5.0
d. Import-Export improvements and error catching
e. bug fixes. 

   We will optimize and put our efforts into this issue as well. Please be patient. 

   **Please keep this thread for beta issues.** 

2. @kangarko  - sorry for that. I will pass it on to the dev & QA team.

Thanks

**Comment by kangarko** [comment]
Thank you, I appreciate it! Lemme know if you need my debug logs or anything.

**Comment by kangarko** [comment]
@DorShahar unfortunately there is a fatal error preventing me from editing a custom post type on the latest beta: ![a](https://i.imgur.com/DI3zzKs.png)

**Comment by kangarko** [comment]
@DorShahar urgent: the beta3 from/releases you just uploaded broke shortcodes and internal URLs: https://i.imgur.com/Y2XQozT.png

**Comment by DorShahar** [comment]
Hey all,
We just released a new beta version, please update your version to **Elementor v3.5.0 Beta 3.**
The new version includes the following:

* Fix: Column padding affected absolute positioned elements (#16572)
* Fix: Some controls disappears in responsive devices when Additional Custom Breakpoint experiment is active (#16608)
* Fix: Empty state background was missing in the Media controls
* Fix: Apply Kit option threw an error with PHP 8.0
* Fix: Sticky option threw an error and cause the editor not to work
* Fix: Removed the hotkeys description from the panel footer tooltips
* Fix: UI glitch in "Get beta updates" modal


Thank you

**Comment by kangarko** [comment]
@DorShahar the new beta 3 has a major bug I reported above

**Comment by kangarko** [comment]
Another problem, my internal error reporting mechanism sent this to me while saving popup:

```
Error: Call to undefined method Elementor\Core\Kits\Documents\Kit::save_display_settings_data()
Trace:

#0 [internal function]: ElementorPro\Modules\Popup\Module->save_display_settings(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#1 /data/a/d/ad84617b-c5b5-4005-b483-a6f3c86c02d6/mineacademy.org/sub/app/wp-content/plugins/elementor/core/common/modules/ajax/module.php(171): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /data/a/d/ad84617b-c5b5-4005-b483-a6f3c86c02d6/mineacademy.org/sub/app/wp-includes/class-wp-hook.php(309): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#3 /data/a/d/ad84617b-c5b5-4005-b483-a6f3c86c02d6/mineacademy.org/sub/app/wp-includes/class-wp-hook.php(337): WP_Hook->apply_filters('', Array)
#4 /data/a/d/ad84617b-c5b5-4005-b483-a6f3c86c02d6/mineacademy.org/sub/app/wp-includes/plugin.php(470): WP_Hook->do_action(Array)
#5 /data/a/d/ad84617b-c5b5-4005-b483-a6f3c86c02d6/mineacademy.org/sub/app/wp-admin/admin-ajax.php(187): do_action('wp_ajax_element...')
#6 {main}

Viewing page: https://app.mineacademy.org/wp-admin/admin-ajax.php

```

**Comment by DorShahar** [comment]
@kangarko , could you please send the Popup template?

**Comment by kangarko** [comment]
Seems like it does not happen again, I had to rollback to beta2 due to the bugs mentioned above. 

There is another error though:

![a](https://i.imgur.com/U2YCWVd.png)

Template is here: 

[elementor-946-2021-11-22.zip](https://github.com/elementor/elementor/files/7578306/elementor-946-2021-11-22.zip)

**Comment by gilesholder** [comment]
Beta 3 link goes to beta 2 changelog.

**Comment by kangarko** [comment]
And another one:

![a](https://i.imgur.com/9pJnkvT.png)

**Comment by DorShahar** [comment]
Thanks @kangarko . We are looking into it.

**Comment by DorShahar** [comment]
Hey all,
We just released a new beta version, please update your version to Elementor **v3.5.0 Beta 4.**
The new version includes the following:
* Tweak: Adjusted Core widgets to work with Inline-CSS Experiment and Additional Custom Breakpoints experiment (#16126)
* Tweak: Changed the maximum offset value to 1000px in Transform controls
* Tweak: Implemented Redux framework 
* Tweak: Allow external extending of element editing handles buttons
* Tweak: Redundant inline CSS code was removed with the use of `eicons`
* Fix: The `<a>` tags weren't created when shortcodes and internal URL's were in use (#16971)
* Fix: Default values in rotate 3D weren't applied in Transfrom
* Fix: Scroll snap caused UI glitches and threw undefined error on Archive pages
* Fix: Vertical align control disappeared if used in a custom position on a custom breakpoint
* Fix: Choose controls in RTL sites were inverted
* Fix: Elementor notices were being displayed in the plugin update screen
* Fix: Z-index control was overrides negative values 
* Fix: Text Path widget causes redundant file system calls
* Fix: Conflict with JetEngine plugin

Note that we have several known bugs. We will fix them in the next beta. 

Thanks

**Comment by kangarko** [comment]
@DorShahar which known bugs please?

**Comment by Kishorchandth** [comment]
Running on the latest Beta build and here are the problems - 

1. none of the templates, blocks, and even my Templates are not loading.
![Console error](https://user-images.githubusercontent.com/39625343/143175420-77b0c0a7-4194-4cfe-887f-cf3973df1844.jpg)

2. Even if I click the "**activate all the experiments**" button, it wouldn't activate anything to fix this issue I have to click the "**Regenerate CSS & Data**" and **"sync library**" and the indicator is green. This can bug someone who just started out. I think it can fix.

3. Fresh install and it keeps asking me to enable "**safe mode**" - 
[system-info-elementor-beta-testing.local-24-11-2021.txt](https://github.com/elementor/elementor/files/7592805/system-info-elementor-beta-testing.local-24-11-2021.txt)
4. Hamburger menu (menu close) it is not fixed yet (**known Issue and already mention**)

**Comment by DorShahar** [comment]
Hi @Kishorchandth 
1. This error is know, and already [merged](https://github.com/elementor/elementor/pull/17019) for the next beta.
2. I couldn't reproduce this problem of activating/deactivating all experiments. In what browser are you using? 
3. As for this problem - I'll consult the developers
4. This will be fixed in a pro version after the core version will be released, due to a depandancy between the versions. 

Hi @kangarko  , The know issues are:

- The template library is unreachable 
- Your report about saving a template that throws an error. We are still looking into that, and we might contact your on this thread 
- Errors in Drag from Desktop feature caused by SVG's and Json's
- Paypal button didn't work 

Thanks

**Comment by kangarko** [comment]
@DorShahar 

![a](https://i.imgur.com/RYD1bNO.png)

This happens on beta4 as well as the latest stable release when adding steps to form fields on a custom form widget. I simply cloned your Form widget into a custom class, left everything as-is and getting this error.

**Comment by DorShahar** [comment]
Hey all,
We just released a new beta version, please update your version to **Elementor v3.5.0 Beta 5.**
The new version includes the following:

* Fix: Unable to access to the Template Library (#17014)
* Fix: Saving a template with a condition threw an error (#16840)
* Fix: Adding a PayPal Button widget threw an error 

Thanks

**Comment by fabianenzensperger** [comment]
> Hey all, We just released a new beta version, please update your version to **Elementor v3.5.0 Beta 5.** The new version includes the following:
> 
> * Fix: Unable to access to the Template Library ([Wordpress - Elementor - Menu "blocks" templates not loading #17014](https://github.com/elementor/elementor/issues/17014))
> * Fix: Saving a template with a condition threw an error ([🚀   Elementor v3.5.0 Beta 5 Release  🚀  #16840](https://github.com/elementor/elementor/issues/16840))
> * Fix: Adding a PayPal Button widget threw an error
> 
> Thanks


@DorShahar 

Are these the several known bugs as you wrote above? And the release date has been postponed, hasn't it?

**Comment by kangarko** [comment]
@DorShahar did you fix the console error when adding steps to forms? I already downgraded due to the amount of bugs, reluctant to test on live environment until a confirmed fix. Thanks!

**Comment by tekonemedia** [comment]
I hope we get perspective settings for css transforms.

**Comment by DorShahar** [comment]
> I hope we get perspective settings for css transforms.

@tekonemedia , You can use perspective in Rotate 3D:
![image](https://user-images.githubusercontent.com/19154938/144748234-96caf98c-abb4-449e-b929-28b8b30220d7.png)

**Comment by tekonemedia** [comment]
That's fantastic!

**Comment by babakfp** [comment]
@fabianenzensperger

> There is a video from Elementor that I think was released in the summer with an announcement of features that should appear in 2021.
> 
> <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">

Where can I find that video?

**Comment by paauserrat** [comment]
> @fabianenzensperger
> 
> > There is a video from Elementor that I think was released in the summer with an announcement of features that should appear in 2021.
> > <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">
> 
> Where can I find that video?

https://www.youtube.com/watch?v=zUBG27llWKQ

**Comment by fabianenzensperger** [comment]
@DorShahar So it is already 1 day later than 3.5 should have been officially released. Is there any information about this?

**Comment by babakfp** [comment]
<img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">

What are "Nested Elements"?

**Comment by sf-steve** [comment]
> <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">
> 
> What is "Nested Elements"?

IMHO the most important feature in the whole list - the ability for one widget to contain other widgets.

For example, the accordion & tab widgets are currently extremely limited, as they can only contain formatted text (or a shortcode). Similar problem for many other built in widgets, and custom developed widgets - since all content must be predefined at dev time, you end up with a lot of duplicated, similar widgets, and overly complicated widget admin UIs (or shortcode hacks that destroy the editing flow)

Nested elements would change this - presumably by adding some kind of dropzone/container control.

This is likely to be challenging to develop, since the Elementor admin UI relies on enhancing the (unknown, controlled by 3rd party theme/plugin code) frontend UI.

Currently, only sections (and containers in the dev version) contain dropzones. These are always visible in the admin UI, so can always be drag and dropped into.

Tabs, accordiants, slider slides however, are not always visible, so a different UI will be needed

**Comment by fabianenzensperger** [comment]
> > <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">
> > What is "Nested Elements"?
> 
> IMHO the most important feature in the whole list - the ability for one widget to contain other widgets.
> 
> For example, the accordion & tab widgets are currently extremely limited, as they can only contain formatted text (or a shortcode). Similar problem for many other built in widgets, and custom developed widgets - since all content must be predefined at dev time, you end up with a lot of duplicated, similar widgets, and overly complicated widget admin UIs (or shortcode hacks that destroy the editing flow)
> 
> Nested elements would change this - presumably by adding some kind of dropzone/container control.
> 
> This is likely to be challenging to develop, since the Elementor admin UI relies on enhancing the (unknown, controlled by 3rd party theme/plugin code) frontend UI.
> 
> Currently, only sections (and containers in the dev version) contain dropzones. These are always visible in the admin UI, so can always be drag and dropped into.
> 
> Tabs, accordiants, slider slides however, are not always visible, so a different UI will be needed


It will definitely be very exciting to see how Elementor implement this function. Currently, the system is not designed for this.

**Comment by Michaelgimii** [comment]
Pro version 3.6, in addition to page transitions, what other features will it come with? Is information available?

**Comment by fabianenzensperger** [comment]
> Pro version 3.6, in addition to page transitions, what other features will it come with? Is information available?


So far, flexbox container too.

**Comment by babakfp** [comment]
> > <img alt="Bildschirmfoto 2021-11-07 um 23 05 54" width="372" src="https://user-images.githubusercontent.com/57997262/140663630-72ff6903-6004-485b-be36-d6cce83f98c2.png">
> > What is "Nested Elements"?
> 
> IMHO the most important feature in the whole list - the ability for one widget to contain other widgets.
> 
> For example, the accordion & tab widgets are currently extremely limited, as they can only contain formatted text (or a shortcode). Similar problem for many other built in widgets, and custom developed widgets - since all content must be predefined at dev time, you end up with a lot of duplicated, similar widgets, and overly complicated widget admin UIs (or shortcode hacks that destroy the editing flow)
> 
> Nested elements would change this - presumably by adding some kind of dropzone/container control.
> 
> This is likely to be challenging to develop, since the Elementor admin UI relies on enhancing the (unknown, controlled by 3rd party theme/plugin code) frontend UI.
> 
> Currently, only sections (and containers in the dev version) contain dropzones. These are always visible in the admin UI, so can always be drag and dropped into.
> 
> Tabs, accordiants, slider slides however, are not always visible, so a different UI will be needed

Ahah. Thank you for your excellent explanation.
I think the Container widget was the first step to implement the Nested Element feature because with sections and columns it wasn't possible to nest them either. You needed to do a hack to nest a subsection inside another subsection...

**Comment by Deadlift79** [comment]
> @DorShahar So it is already 1 day later than 3.5 should have been officially released. Is there any information about this?

Now I read: "December 15, 2021 (Estimated)"

**Comment by black-eye** [comment]
Hi,

I've came accross quite important change in legacy css file "elementor\assets\css\frontend-legacy.css". For some reason (I don't understand) the predefined column padding settings were moved from `.elementor-element-populated` to it's child `.elementor-element-populated > .elementor-widget-wrap`.

Why have you done this? I would expect legacy mode should provide flaweless experience for older sites, but this one broke all my custom settings. BTW, using those settings I tried to deal with this problem: https://github.com/elementor/elementor/issues/452. Unfortunatelly, every new update makes my custom solution more and more complicated: first you have removed the `.elementor-row` element, but at least there was the legacy mode. Now even the legacy mode is broken.

Is there a reason for this or is this some kind of overlook? Thanks for the info.

![obrazek](https://user-images.githubusercontent.com/5634870/144911773-8f2313a9-c237-4430-bae5-4607e35f67b4.png)

**Comment by fabianenzensperger** [comment]
> > @DorShahar So it is already 1 day later than 3.5 should have been officially released. Is there any information about this?
> 
> Now I read: "December 15, 2021 (Estimated)"


Ok, thanks, unfortunately in the last few weeks of a beta we usually don't get any more regular information from the developers. 🙋‍♂️

Well, they must have a lot to do.. 👍

**Comment by ROGMARQ14** [comment]
Doesn´t converting eicons into inline SVG increase the size of the DOM?

**Comment by DorShahar** [comment]
> Hi,
> 
> I've came accross quite important change in legacy css file "elementor\assets\css\frontend-legacy.css". For some reason the predefined column padding settings were moved from `.elementor-element-populated` to it's child `.elementor-element-populated > .elementor-widget-wrap`.
> 
> Why have you done this? I would expect legacy mode should provide flaweless experience for older sites, but this one broke all my custom settings. BTW, using those settings I tried to deal with this problem: #452. Unfortunatelly, every new update makes my custom solution more and more complicated: first you have removed the `.elementor-row` element, but at least there was the legacy mode. Now even the legacy mode is broken.
> 
> Is there a reason for this or is this some kind of overlook? Thanks for the info.
> 
> ![obrazek](https://user-images.githubusercontent.com/5634870/144911773-8f2313a9-c237-4430-bae5-4607e35f67b4.png)

HI @black-eye 
This fix was part of:
- Fix: Column padding affected absolute positioned elements (Elementor v 3.4.5 caused absolutely positioned items to move slightly out of position #16572)

You are correct - and indeed we need to give an option for backward compatibility.

--- 


> Doesn´t converting eicons into inline SVG increase the size of the DOM?

Hi @ROGMARQ14  - this is an experiment. You can activate or deactivate it. There are pros and cons for each side.

**Comment by zeinnicholas** [comment]
This update seems to introduce a lot of changes, and many hard deprecated (removed) code. Please, before releasing or pushing the update make sure to prepare the public accordingly by releasing a post way before the version is available, explaining what is new and how people can prepare for this update to avoid the obvious public outrage when things go wrong. Especially since many users have add-ons installed that will probably break with this release if they aren't compatible. Let's avoid doing the same mistakes of the past ok? This will be highly appreciated!

**Comment by Kishorchandth** [comment]
> Doesn´t converting eicons into inline SVG increase the size of the DOM?

Yes, it will increase DOM size a bit but you also save 80KB from eicons, more than 150KB from FontAwesome icons.

**Comment by black-eye** [comment]
> HI @black-eye This fix was part of:
> 
>     * Fix: Column padding affected absolute positioned elements (Elementor v 3.4.5 caused absolutely positioned items to move slightly out of position

Well, by inspecting the original problem, it seems that in 4.4.5 you've released a "feature", which broke the existing absolute possitioned layouts. So you have decided to fix this issue by breaking yet another part of the design. What was the purpose of the original change in the first place? There is no reply from the devs in the original thread to this question and the users there don't seem to understand why this change has been introduced and why it wasn't mentioned in the change log. Wouldn't be the best solution to roll back the change in 4.4.5? 

Maintaing Elementor sites has become huge pain, especially when you've done this big mistake of customizing it to your needs. It's like patching the patches that fixes the patches over another patches. Unfortunately we need these custom enhancements to achieve basic stuff which isn't provided by default.

**Comment by DorShahar** [comment]
Hi @zeinnicholas  - regarding your comment. For each version, we are releasing deprecations blog post. Also to this one. See [this link](https://github.com/elementor/elementor/issues/16840#issuecomment-987833263) at the end of the first post (end of the changelog). 
In case that a function was deleted and we didn't declare it, please let us know! 

---


@black-eye  - the fix was part of the Rescue DOM elements experiment. During the experiment, we harmed sites in v3.4.5, and therefore, we fixed it. 
We aren't running away from responsibility, and are doing our best to improve our quality and backwards compatibility to any feature.

**Comment by DorShahar** [comment]
Hi All,
We just released a new beta version, please update your version to **Elementor v3.5.0 Beta 6**
Due to maintenance we made to our release system the version **you are receiving has a larger increment than planned** (beta8 instead of beta6).

This version is the release candidate (RC) and includes the following change:
- Tweak: Reverted Button trait module update

Thanks

**Comment by zeinnicholas** [comment]
> Hi @zeinnicholas  - regarding your comment. For each version, we are releasing deprecations blog post. Also to this one. See [this link](https://github.com/elementor/elementor/issues/16840#issuecomment-987833263) at the end of the first post (end of the changelog). 
> In case that a function was deleted and we didn't declare it, please let us know! 
> 

Hello @DorShahar thanks for the response. 

Let me be more clear, perhaps you didn't understand what I meant:

At this point, it's common practice of Elementor to push releases (especially Core releases) before making any announcement to the general public. The releases are pushed by Sunday and an announcement is only made in Elementor's blog 3 to 4 days later.

For us, beta testers, we are aware of the changes. But for the general public this adds confusion, followed by frustration when things go wrong. 

I'm asking if you can please do the other way around:

1. First release a blog post about the upcoming release and what might change, explaining how to proceed before updating (making backups, testing first, clearing cache, making sure add-ons are compatible, etc). 

2. Make sure the changelog is updated in the WordPress repository (it never is in time). 

3. Push the release. 

4. Announce the new version has been released and appease expectations (something like "it may take some time for the update to appear in your Dashboard"). Reinforcing the need to backup, test it out, clear cache, and to make sure add-ons are compatible.


This new version will inevitably break some sites, most likely because of cache, but probably also because of incompatible add-ons. 

I'm asking you to be preemptive. Not just for devs like us, but to the general public. 

Thanks for your hard work.

**Comment by DorShahar** [comment]
Hi @zeinnicholas  - thanks for clarifying this one!
We will examine our process again so we could improve it.

**Comment by fabianenzensperger** [comment]
So 3.5.0 is rolling out? 🤗

**Comment by DorShahar** [comment]
Hello All!

Core v3.5.0 has been officially released!
Thank you for all of your feedback and efforts.

Thank you all!

**Comment by fabianenzensperger** [comment]
> Hello All!
> 
> Core v3.5.0 has been officially released! Thank you for all of your feedback and efforts.
> 
> Thank you all!


@DorShahar With pleasure, I'm looking forward to 3.6! Can we expect a beta this year?

What happens to the rest of the things that have been announced for 2021? Will they come in 2022? I mean Mega menu, Nested Elements, Flexbox Container, Elementor Presets and Page transitions..

**Comment by babakfp** [comment]
What happened to the Container widget? It's not available in 3.5.0🤨

**Comment by fabianenzensperger** [comment]
> What happened to the Container widget? It's not available in 3.5.0🤨


It's not in 3.5! it's in dev Beta 3.6.0

**Comment by caspervoogt** [comment]
I'm using Elementor Pro 3.6.5 including dev edition, and have also tried 3.7.0-dev34 with 'Elementor Beta (Developer Edition)', and I don't see the Container element options anywhere.
